tests/test_io/conftest.py

```
@@ -0,0 +1,134 @@
+    sock = config.bind_socket()
+    host, port = sock.getsockname()[:2]
+
+    def _run():
+        server.run(sockets=[sock])
+
+    thread = threading.Thread(target=_run, daemon=True)
+    thread.start()
+    probe_url = f"http://{host}:{port}/das/example_dasdae_event_1.h5"
+    for _ in range(50):
+        try:
+            with urlopen(probe_url):
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)